pylint/checkers/non_ascii_names.py

```diff
@@ -0,0 +1,264 @@
+# Copyright (c) 2021-2022 Carli Freudenberg <carli.freudenberg@energymeteo.de>
+
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+"""All alphanumeric unicode character are allowed in Python but due
+to similarities in how they look they can be confused.
+
+See: https://www.python.org/dev/peps/pep-0672/#confusable-characters-in-identifiers
+
+The following checkers are intended to make users are aware of these issues.
+"""
+
+import re
+import sys
+from typing import Optional, Union
+
+from astroid import nodes
+
+import pylint.checkers.base
+import pylint.checkers.utils
+from pylint import interfaces
+from pylint.constants import HUMAN_READABLE_TYPES
+from pylint.lint import PyLinter
+
+if sys.version_info >= (3, 8):
+    from typing import Protocol
+else:
+    from typing_extensions import Protocol
+
+
+if sys.version_info[:2] >= (3, 7):
+    # pylint: disable-next=fixme
+    # TODO: Remove after 3.6 has been deprecated
+    Py37Str = str
+else:
+
+    class Py37Str(str):
+        # Allow Python 3.6 compatibility
+        def isascii(self: str) -> bool:
+            return all("\u0000" <= x <= "\u007F" for x in self)
+
+
+class _AsciiOnlyCheckedNode(Protocol):
+    _is_ascii_only: bool
+
+
+class NonAsciiNamesChecker(
+    pylint.checkers.BaseChecker, pylint.checkers.base.NameCheckerHelper
+):
+    """A strict name checker only allowing ASCII
+
+    If your programming guideline defines that you are programming in English,
+    then there should be no need for non ASCII characters in Python Names.
+    Everybody else can simply not use this check.
+
+    Note: This check only checks Names, so it ignores the content of
+          docstrings and comments!
```

Comment:
This should go in the doucmentation.
Actually what I am missing is a part of the User documentation that explain/lists all checkers/messages and possibly all configuration Options...
(Best of course auto generated)
Or did I miss this @DanielNoord ?


Currently using pylint is a bit overwhelming in this regards.

I looked into the message and I think they are to the point. This is really just an explaination for the programmers why we need this checker at all.

Reply:
I have been working on this in #5396.

Currently the only way to store extra data per message is to store it in the message description, which is what that PR takes to populate each message's page. Also see follow-up #5527 which discuss better ways to store additional data per message. Perhaps for now we can add it to the message description as it would then be added to the documentation when #5396 gets merged?

Reply:
Integrated